completion/models.py

```diff
@@ -201,6 +201,19 @@ def user_course_completion_queryset(cls, user, course_key):
         """
         return cls.objects.filter(user=user, course_key=course_key)
 
+    @classmethod
+    def get_last_sitewide_block_completed(cls, user):
+        """
+        query latest completion for user (any course)
+        Return value:
+            obj: block completion
+        """
+        try:
+            latest_block_completion = cls.objects.filter(user=user).latest()
```

Comment:
This won't work with WL sites in the mix.
You can tell whether the course belongs to a certain site based on the organization encoded into the course_id. Organizations have a many to 1 relationship to a site. The organizations associated with a white label site is at "Site Configuration" on LMS. For example: at https://courses.edx.org/admin/site_configuration/siteconfiguration/7/, look at the field `course_org_filter`.
The solution would be, to retrieve, for the learner, a dictionary of course mapped to last completed block, and let the consumer code to filter based on the org aforementioned